src/platform/h563xx/adc_ll.c

```diff
@@ -22,6 +22,11 @@
 
 enum { ADC_IRQ_PRIORITY = 1 }; // ADC interrupt priority
 
+bool g_adc1_initialized = false; // Flag to indicate if ADC1 is initialized
+bool g_adc2_initialized = false; // Flag to indicate if ADC2 is initialized
+bool g_adc_multimode =
+    true; // Flag to indicate if ADC1 and ADC2 are in multimode
```

Comment:
I think this is the main thing that's preventing this from working. You're enabling multimode by default, but you're not unsetting this if another mode is chosen. Thus, when trying to run in single ADC-mode, you're actually running some kind of in-between mode.

After changing this (and a few other things, but I think was the main thing), I get output once. Then it errors out during ADC_restart, so there is some problem there as well. I don't have time to look into that tonight, will try to do so tomorrow.